Common/Python/PandasConverter.cs

```diff
@@ -346,7 +387,18 @@ public PyObject ToPandasDataFrame(dynamic pandas, int levels = 2)
                 var index = pandas.MultiIndex.from_tuples(tuples, names: names.Split(','));
 
                 // Returns a dictionary keyed by column name where values are pandas.Series objects
-                return _series.ToDictionary(k => k.Key, v => pandas.Series(v.Value, index));
+                var pandasSeries = new Dictionary<string, dynamic>();
+                foreach (var kvp in _doubleSeries)
+                {
+                    if (kvp.Value.Count != tuples.Length) continue;
+                    pandasSeries.Add(kvp.Key, pandas.Series(kvp.Value, index));
+                }
+                foreach (var kvp in _stringSeries)
+                {
```

Comment:
Instead of having a string series and a double series, we could define the series to have values of type object.  This maybe be especially relevant when decomposing dynamic data type's storage dictionary since we don't know what the property type would be on the custom data.